pkg/operator/controller/status/controller.go

```diff
@@ -392,11 +407,72 @@ func computeOperatorUpgradeableCondition(ingresses []operatorv1.IngressControlle
 		cond := nonUpgradeableIngresses[ingress]
 		message = fmt.Sprintf("%s ingresscontroller %q is not upgradeable: %s: %s", message, ingress.Name, cond.Reason, cond.Message)
 	}
+
+	return false, message
+}
+
+// checkGatewayAPICRDs will check if Gateway API CRDs exist and if they are compatible
+// TODO: Consider getting the managedCRDs from the gatewayAPI controller? Currently duplicates the list from there.
+// TODO: validate existing gatewayapi crds with crd-schema-checker
+func checkGatewayAPICRDs(gatewayAPICRDs []apiextensionsv1.CustomResourceDefinition) (bool, string) {
+	var managedGatewayAPICRDs = []*apiextensionsv1.CustomResourceDefinition{
+		manifests.GatewayClassCRD(),
+		manifests.GatewayCRD(),
+		manifests.HTTPRouteCRD(),
+		manifests.ReferenceGrantCRD(),
+	}
+
+	managedCRDNames := map[string]bool{}
```

Comment:
Just a nit remark. The optimal (and I believe go-idiomatic too) way to do a set data structure is to use an empty `struct`, this way you don't waste bytes for storing `bool`s.
```suggestion
	managedCRDNames := map[string]struct{}{}
```

Reply:
This now evolved into a map of pointers to the crds.